src/dev/typescript/run_type_check_cli.ts

```diff
@@ -87,5 +87,10 @@ export function runTypeCheckCli() {
     process.exit(1);
   }
 
-  execInProjects(log, projects, 'tsc', project => ['--project', project.tsConfigPath, ...tscArgs]);
+  execInProjects(log, projects, process.execPath, project => [
+    ...(project.name === 'x-pack' ? ['--max-old-space-size=2048'] : []),
```

Comment:
I think we can start like that, and in case some other projects starting to have OOM, I think we can just define the `NODE_OPTIONS` env with that flag enabled for every project here https://github.com/elastic/kibana/blob/0956f951dc2378c43f11ac38e3877d4894d8c884/src/dev/typescript/exec_in_projects.ts#L44

Reply:
My only concern is that running this for multiple projects means that we need multiples of 2GB of RAM available just to run the type check. I'd rather whitelist projects and then we can take down the concurrency further if necessary.

Reply:
Yeah agreed!